app/Database/InProgCourseQueries.hs

```diff
@@ -0,0 +1,70 @@
+--| UberGoal: go from T.Text -> [Entity Lecture] / [Entity Tutorial] without multiple DB queries.
+
+import Data.Tuple.Select as Tup -- selecting first, second and third elements of tuples (fall, spring, year)
+
+-- | Step 1: single query for lecture or tutorial. Produces list of all 
+-- | lectures/tutorials respectively, whcih are not separated by session
+
+-- | Queries db for all matching records with lecture or tutorial code of this course
+lectureQuery :: T.Text <- [Entity Lecture]
+lectureQuery courseCode = selectList [LectureCode ==. courseCode] []
+
+tutorialQuery :: T.Text <- [Entity Tutorial]
+tutorialQuery courseCode = selectList [TutorialCode ==. courseCode] []
+
+
+-- | Step 2: find a way to split these into separate lists.
+
+-- | goal: refactor both below to something like
+-- | splitSessions :: Entity => [a] -> ([a], [a], [a]) 
+
+-- | Returns a 3-element tuple representing Fall, Spring, and Yearlong entity lists.
+-- | problem: Data.Text vs String
+splitSessionsL :: [Entity Lecture] -> ([Entity Lecture], [Entity Lecture], [Entity Lecture])
+splitSessionsL lecturesList = do
```

Comment:
Only use `do` notation for monads - the only monads you care about right now are `IO` and `ReaderT SqlBackend`.

In this case, you can just use a simple `let` to bind `fallLec = getBySessionL lecturesList "F"`, etc.